pkg/assembler/backends/testing/artifact.go

```diff
@@ -17,67 +17,154 @@ package testing
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strconv"
 	"strings"
 
+	"github.com/vektah/gqlparser/v2/gqlerror"
+
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
 )
 
-func registerAllArtifacts(client *demoClient) {
-	// strings.ToLower(string(checksum.Algorithm)) + ":" + checksum.Value
-	client.registerArtifact("sha256", "6bbb0da1891646e58eb3e6a63af3a6fc3c8eb5a0d44824cba581d2e14a0450cf")
-	client.registerArtifact("sha1", "7A8F47318E4676DACB0142AFA0B83029CD7BEFD9")
-	client.registerArtifact("sha512", "374AB8F711235830769AA5F0B31CE9B72C5670074B34CB302CDAFE3B606233EE92EE01E298E5701F15CC7087714CD9ABD7DDB838A6E1206B3642DE16D9FC9DD7")
+// Internal data: Artifacts
+type artMap map[string]*artStruct
```

Comment:
Is the map keyed on the algorithm or on the digest?

Reply:
i think it is keyed on both "alg:digest"?

Reply:
yep, both.